prompt2model/dataset_generator/prompt_based.py

```diff
@@ -32,18 +32,27 @@
 
 @dataclass(frozen=True)
 class Example:
-    """An example from a dataset, containing input and output columns."""
+    """An example from a dataset, containing input, explanation and output columns."""
 
     input_col: str
+    explain_col: str
```

Comment:
I would suggest that we just call this `explanation_col` to be consistent. Also, it's not clear what an explanation is. I'd suggest that you expand the docstring and describe what all the variables are.

```suggestion
    explanation_col: str
```